File: inspect_geojson.py
import json
import os

file_path = r"c:\Users\haona\OneDrive\Desktop\Viet_Map\assets\geo\communes.geojson"

try:
    with open(file_path, 'r', encoding='utf-8') as f:
        # We don't want to load the whole 178MB if we can avoid it, 
        # but json.load is easiest if memory allows. 178MB is fine for python.
        data = json.load(f)
        
    print(f"Type: {data.get('type')}")
    features = data.get('features', [])
    print(f"Number of features: {len(features)}")
    
    if features:
        print("\nProperties of the first feature:")
        print(json.dumps(features[0].get('properties', {}), indent=2, ensure_ascii=False))
        
except Exception as e:
    print(f"Error: {e}")

File: inspect_geojson.dart
import 'dart:convert';
import 'dart:io';

void main() async {
  final file = File(r'c:\Users\haona\OneDrive\Desktop\Viet_Map\assets\geo\communes.geojson');
  
  if (!await file.exists()) {
    print('File not found');
    return;
  }
  
  try {
    print('Reading file...');
    String stringData = await file.readAsString();
    print('Fixing NaN...');
    // Replace NaN (not surrounded by quotes) with null.
    // A simple replaceAll might catch letters in strings, but we know it's `: NaN`
    stringData = stringData.replaceAll(': NaN', ': null');
    
    print('Decoding JSON...');
    final data = jsonDecode(stringData);
    
    final features = data['features'] as List;
    print('Number of features: ${features.length}');
    
    if (features.isNotEmpty) {
      print('\nProperties of the first feature:');
      print(JsonEncoder.withIndent('  ').convert(features[0]['properties']));
    }
  } catch (e) {
    print('Error: $e');
  }
}
